packages/mui-base/src/FormControlUnstyled/FormControlUnstyled.tsx

```diff
@@ -96,6 +96,7 @@ const FormControlUnstyled = React.forwardRef(function FormControlUnstyled<
 
   const [focused, setFocused] = React.useState(false);
   if (disabled && focused) {
+    // NOTE: Setting state during render?
     setFocused(false);
   }
```

Comment:
Yeah, this should be in a useEffect. Good catch.

```suggestion
  React.useEffect(() => {
    if (disabled && focused) {
      setFocused(false);
    }
  }, [disabled, focused]);
```

Reply:
Opened a separate PR https://github.com/mui/material-ui/pull/34916